compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -287,16 +291,31 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
             VarValue::ErrorValue => return false,
             VarValue::Value(r) => r,
         };
+        debug!("enforce_member_constraint: lower_bound={:#?}", member_lower_bound);
+        if member_constraint.choice_regions.contains(&member_lower_bound) {
+            debug!("enforce_member_constraint: lower bound is already a valid choice");
+            return false;
+        }
 
         // Find all the "upper bounds" -- that is, each region `b` such that
         // `r0 <= b` must hold.
         let (member_upper_bounds, ..) =
             self.collect_bounding_regions(graph, member_vid, OUTGOING, None);
+        debug!("enforce_member_constraint: upper_bounds={:#?}", member_upper_bounds);
+
+        // If there are no upper bounds, and static is a choice (in practice, it always is),
+        // then we should just pick static.
+        if member_upper_bounds.is_empty()
```

Comment:
I notice the check here is simpler than the one with `any_upper_bounds` done above (_e.g._, non-empty case with only `'static`s inside): does `collect_bounding_regions` never yield `'static`s?

Reply:
Conjecture on my part: I wouldn't expect it to ? It wouldn't constrain the variable to add it as an upper bound, it's already the `⊤` element. 

We'll see when we do the Zoom session ^^

Reply:
Was the zoom session recorded somewhere btw?

Reply:
It'll [likely happen tomorrow](https://rust-lang.zulipchat.com/#narrow/stream/187312-wg-async-foundations/topic/member.20constraints.20PR.20.2389056/near/254337073)